### ISSUE TITLE
Incorrect work in modal window

### ISSUE THREAD

**Issue by svk-man** [abiodunolunu/vue3-star-ratings#5]
Hi! Cool plugin! 😎 

But unfortunately I get a rating calculation error when using the plugin in a modal window.

![image](https://github.com/abiodunolunu/vue3-star-ratings/assets/17108861/e5466d10-9a18-45d9-befc-0f2c2a6fbe3d)

I have prepared a [repository](https://github.com/svk-man/stars-test) where I reproduced the error.
I also used [an example of the modal form](https://vue-final-modal.org/use-cases/modal-login-form).

Please fix it

**Comment by abiodunolunu** [comment]
Thanks, @svk-man  for catching this issue. I didn't realize how complicated things could get when our component is nested inside parent elements with non-static positioning, especially those with absolute positioning. These types of positioning contexts can really change the size and position of our element.

When I took a closer look, I saw that I was miscalculating the click position relative to our star component. The original code used offsetLeft and offsetWidth, which work fine when no external positioning contexts affect our component. But when parent elements have specific CSS positions, like absolute, these values can be misleading.

To fix this, I switched to using the getBoundingClientRect() method, which gives us dimensions and positions relative to the viewport. Now, I can calculate the relative X position using pageX - rect.left and the width with rect.width. This ensures accurate behavior, no matter what the parent's positioning context is.

Thanks for your patience and feedback. It's been really helpful in refining the component. Your issue has helped me improve the accuracy and reliability of the star rating component. 

Fixed in: 3.0.4
